CraftTweaker2-API/src/main/java/crafttweaker/api/data/DataFloat.java

```diff
@@ -17,7 +17,7 @@ public DataFloat(float value) {
     
     @Override
     public boolean asBool() {
-        throw new IllegalDataException("Cannot cast a float to a bool");
+        return value != 0.0f;
```

Comment:
I feel this should better mimic how Java does it, which is "if this isn't true (or 1), then it should be considered false).

So change this to `value == 1.0f`